src/app/components/confirmBtcTransactionComponent/inputOutputComponent.tsx

```diff
@@ -168,22 +169,42 @@ function InputOutputComponent({ address, parsedPsbt, isExpanded, onArrowClick }:
           <OutputTitleText>{t('OUTPUT')}</OutputTitleText>
           {parsedPsbt?.outputs.map((output) => (
             <TransferDetailContainer>
-              <TransferDetailView
-                icon={OutputIcon}
-                hideAddress
-                hideCopyButton={btcAddress === output.address || ordinalsAddress === output.address}
-                amount={`${satsToBtc(new BigNumber(output ? output.amount : 0n)).toString()} BTC`}
-                address={output.address}
-              >
-                {output.address === btcAddress || output.address === ordinalsAddress ? (
-                  <TxIdContainer>
-                    <YourAddressText>(Your Address)</YourAddressText>
+              {output.outputScript ? (
+                <TransferDetailView
+                  icon={ScriptIcon}
+                  hideAddress
+                  hideCopyButton={true}
+                  amount={`${satsToBtc(
+                    new BigNumber(output ? output.amount.toString() : '0'),
+                  ).toString()} BTC`}
+                  address={output.address}
+                  outputScript={output.outputScript}
+                  outputScriptIndex={scriptOutputCount}
+                >
+                  <SubValueText>{`${t('SCRIPT_OUTPUT')} #${scriptOutputCount++}`}</SubValueText>
+                </TransferDetailView>
+              ) : (
+                <TransferDetailView
+                  icon={OutputIcon}
+                  hideAddress
+                  hideCopyButton={
+                    btcAddress === output.address || ordinalsAddress === output.address
+                  }
+                  amount={`${satsToBtc(
+                    new BigNumber(output ? output.amount.toString() : '0'),
+                  ).toString()} BTC`}
+                  address={output.address}
+                >
+                  {output.address === btcAddress || output.address === ordinalsAddress ? (
+                    <TxIdContainer>
+                      <YourAddressText>(Your Address)</YourAddressText>
+                      <SubValueText>{getTruncatedAddress(output.address)}</SubValueText>
+                    </TxIdContainer>
+                  ) : (
                     <SubValueText>{getTruncatedAddress(output.address)}</SubValueText>
-                  </TxIdContainer>
-                ) : (
-                  <SubValueText>{getTruncatedAddress(output.address)}</SubValueText>
-                )}
-              </TransferDetailView>
+                  )}
+                </TransferDetailView>
+              )}
```

Comment:
this nested ternary logic is not ideal in the render block.
for better readability, could we shift out the props which require the ternary logic out above the render, so that your final render looks something like this:

```
const detailViewIcon = output.outputScript ? ScriptIcon : OutputIcon;
const detailViewHideCopyButton = output.outputScript || btcAddress === output.address || ordinalsAddresss === output.address;
const detailViewValue = ...

return (
...
<TransferDetailView
  icon={detailViewIcon}
  hideAddress
  hideCopyButton={detailViewHideCopyButton}
  amount={`${satsToBtc(
    new BigNumber(output ? output.amount.toString() : '0'),
  ).toString()} BTC`}
  address={output.address}
  outputScript={output.outputScript}
  outputScriptIndex={scriptOutputCount}
>
  <SubValueText>{detailViewValue}</SubValueText>
</TransferDetailView>
```

Reply:
done . please check